plugins/accelerated-peft/src/fms_acceleration_peft/framework_plugin_autogptq.py

```diff
@@ -321,8 +321,7 @@ def augmentation(
             train_mode=True,  # install adapaters for training
         )
 
-        # Reassign the attributes after PEFT installation replaces the top-level class
-        setattr(model, "is_loaded_in_4bit", True)
+        # Reassign `quantization_method` after PEFT installation replaces the top-level class
```

Comment:
I thought that it will also be the case that `is_loaded_in_4bit` be also dropped. Then can for consistency we should just set it back also

Reply:
Setting the `is_loaded_in_4bit` attribute will cause an error in `accelerator.prepare_model` just before the FSDP wrapping. FSDP will eventually handle the device placements later on so not sure if this attribute is useful in the GPTQ-LoRA case.

```
  File "/data/aaron/experimental/fms-hf-tuning/tuning/sft_trainer.py", line 360, in train
    trainer.train()
  File "/data/aaron/experimental/fms-acceleration/.tox/run-benches/lib/python3.10/site-packages/trl/trainer/sft_trainer.py", line 450, in train
    output = super().train(*args, **kwargs)
  File "/data/aaron/experimental/fms-acceleration/.tox/run-benches/lib/python3.10/site-packages/transformers/trainer.py", line 1938, in train
    return inner_training_loop(
  File "/data/aaron/experimental/fms-acceleration/.tox/run-benches/lib/python3.10/site-packages/transformers/trainer.py", line 2085, in _inner_training_loop
    self.model = self.accelerator.prepare(self.model)
  File "/data/aaron/experimental/fms-acceleration/.tox/run-benches/lib/python3.10/site-packages/accelerate/accelerator.py", line 1311, in prepare
    result = tuple(
  File "/data/aaron/experimental/fms-acceleration/.tox/run-benches/lib/python3.10/site-packages/accelerate/accelerator.py", line 1312, in <genexpr>
    self._prepare_one(obj, first_pass=True, device_placement=d) for obj, d in zip(args, device_placement)
  File "/data/aaron/experimental/fms-acceleration/.tox/run-benches/lib/python3.10/site-packages/accelerate/accelerator.py", line 1188, in _prepare_one
    return self.prepare_model(obj, device_placement=device_placement)
  File "/data/aaron/experimental/fms-acceleration/.tox/run-benches/lib/python3.10/site-packages/accelerate/accelerator.py", line 1425, in prepare_model
    raise ValueError(
ValueError: You can't train a model that has been loaded in 8-bit precision on a different device than the one you're training on. Make sure you loaded the model on the correct device using for example `device_map={'':torch.cuda.current_device()}` or `device_map={'':torch.xpu.current_device()}`
```

Reply:
ok then can you put a comment 

```We do not set is_loaded_in_4bit at this point because otherwise `accelerate.prepare_model` will think the device placement is finalized for the quantized model, and will raise```